### PR TITLE
Kill Seperated Mindshield Icons

### PR DESCRIPTION
**Pull request by Golinth**
## About the PR

Changes mindshields to be combined with job icons, rather than separated. This is how SS13 does it, the only remaining step is to animate the icon.

## Why / Balance

SS13 parity, but the real reason is that I assumed the mindshield icon to be temporary (the original PR was literally called Ghetto Mindshield Icon), and I think this looks a lot better to stop UI bloat, especially after/if [25192](https://github.com/space-wizards/space-station-14/pull/25192) gets merged.

## Technical details

Added a new enum in StatusIconPrototype that defines layers the status icon can have. It is defaulted to the Base layer, so no changes need to be made to accommodate it. Controlled with the "layer:" datafield in a yml file.

## Media

New overlay in action
secHUDtest

It works with multiple HUDs!
omniHUDtest

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:

- tweak: Mindshields are no longer separate and appear as a border around job icons on the SecHUD

### CODE REVIEW

**Comment by UbaserB** [comment]
i like it separated more

**Comment by EmoGarbage404** [comment]
> i like it separated more

Its better combined considering its not super context sensitive information and remains fairly constant for players.

File: Content.Shared/StatusIcon/StatusIconPrototype.cs [hunk 1]
```diff
@@ -34,6 +34,12 @@ public int CompareTo(StatusIconData? other)
     {
         return Priority.CompareTo(other?.Priority ?? int.MaxValue);
     }
+
+    /// <summary>
+    /// The layer the icon is displayed on. Mod is drawn above Base. Base | Mod
+    /// </summary>
+    [DataField("layer")]
```
Comment (EmoGarbage404) [inline Content.Shared/StatusIcon/StatusIconPrototype.cs:41]:
```suggestion
    [DataField]
```

**Review by EmoGarbage404 — APPROVED** [review]

**Comment by Unbelievable-Salmon** [comment]
I hate to be that person, but this seems like it has potential to be a struggle for accessibility. With very poor sight iit takes more effort to look at the HUD icon and be sure of what it actually is. Especially if it's already on a blue job icon. Not saying for certain, but that's just my immediate thoughts.

**Comment by Varen** [comment]
This looks far too subtle an effect for accessibility purposes to be quite honest. Separate symbols make it obvious at a glance even with people who have color blindneeeeeaaannd emo merged it while I was typing

**Comment by Golinth** [comment]
> I hate to be that person, but this seems like it has potential to be a struggle for accessibility. With very poor sight iit takes more effort to look at the HUD icon and be sure of what it actually is. Especially if it's already on a blue job icon. Not saying for certain, but that's just my immediate thoughts.

You're right, and I plan on making a PR that animates the icon TG style, making it flash. You think that would solve accessibility issues by making it more noticeable?

**Comment by Unbelievable-Salmon** [comment]
> > I hate to be that person, but this seems like it has potential to be a struggle for accessibility. With very poor sight iit takes more effort to look at the HUD icon and be sure of what it actually is. Especially if it's already on a blue job icon. Not saying for certain, but that's just my immediate thoughts.
> 
> You're right, and I plan on making a PR that animates the icon TG style, making it flash. You think that would solve accessibility issues by making it more noticeable?

Thank you for looking into this, I appreciate it. To answer you, it depends on how it's animated. If it's subtle it's not going to really be of use. If it's blatant it could work, but that might annoy players. In #accessibility Bhijn and Myr thought about making whether the icon is separated or not toggalablr which might be the best way. 

Either way, I'm sorry that I'm kind of being an inconvenience.

**Comment by Golinth** [comment]
> Either way, I'm sorry that I'm kind of being an inconvenience.

Don’t be. Even though it’s a meme gif, the HoS in this gif has the exact type of flashing that will be added. Subtle enough that it’s not annoying, but also enough difference that it’s easy to tell that that someone has a shield. 
7DDD3AAA-4B16-4361-8B2F-8EA0A5224F62

I was planning on doing it anyway, it’s just now there’s a legitimate reason too

**Pull request merged**